qiskit/circuit/quantumcircuit.py

```diff
@@ -1533,6 +1534,381 @@ def _rebind_definition(self, instruction, parameter, value):
+        from .library.standard_gates.swap import SwapGate
+        return self.append(SwapGate(), [qubit1, qubit2], [])
+
+    def iswap(self, qubit1, qubit2):
+        """Apply :class:`~qiskit.circuit.library.standard_gates.iSwapGate`."""
+        from .library.standard_gates.iswap import iSwapGate
+        return self.append(iSwapGate(), [qubit1, qubit2], [])
+
+    @deprecate_arguments({'ctl': 'control_qubit',
+                          'tgt1': 'target_qubit1',
+                          'tgt2': 'target_qubit2'})
+    def cswap(self, control_qubit, target_qubit1, target_qubit2, *, label=None, ctrl_state=None,
+              ctl=None, tgt1=None, tgt2=None):  # pylint: disable=unused-argument
+        """Apply :class:`~qiskit.circuit.library.standard_gates.CSwapGate`."""
+        from .library.standard_gates.swap import CSwapGate
+        return self.append(CSwapGate(label=label, ctrl_state=ctrl_state),
+                           [control_qubit, target_qubit1, target_qubit2], [])
+
+    @deprecate_arguments({'ctl': 'control_qubit',
+                          'tgt1': 'target_qubit1',
+                          'tgt2': 'target_qubit2'})
+    def fredkin(self, control_qubit, target_qubit1, target_qubit2,
+                *, ctl=None, tgt1=None, tgt2=None):  # pylint: disable=unused-argument
+        """Apply :class:`~qiskit.circuit.library.standard_gates.CSwapGate`."""
+        return self.cswap(control_qubit, target_qubit1, target_qubit2)
+
+    @deprecate_arguments({'q': 'qubit'})
+    def t(self, qubit, *, q=None):  # pylint: disable=invalid-name,unused-argument
+        """Apply :class:`~qiskit.circuit.library.standard_gates.TGate`."""
+        from .library.standard_gates.t import TGate
+        return self.append(TGate(), [qubit], [])
+
+    @deprecate_arguments({'q': 'qubit'})
+    def tdg(self, qubit, *, q=None):  # pylint: disable=unused-argument
+        """Apply :class:`~qiskit.circuit.library.standard_gates.TdgGate`."""
+        from .library.standard_gates.t import TdgGate
+        return self.append(TdgGate(), [qubit], [])
+
+    @deprecate_arguments({'q': 'qubit'})
+    def u1(self, theta, qubit, *, q=None):  # pylint: disable=invalid-name,unused-argument
+        """Apply :class:`~qiskit.circuit.library.standard_gates.U1Gate`."""
+        from .library.standard_gates.u1 import U1Gate
+        return self.append(U1Gate(theta), [qubit], [])
+
+    @deprecate_arguments({'ctl': 'control_qubit',
+                          'tgt': 'target_qubit'})
+    def cu1(self, theta, control_qubit, target_qubit, *, label=None, ctrl_state=None,
+            ctl=None, tgt=None):  # pylint: disable=unused-argument
+        """Apply :class:`~qiskit.circuit.library.standard_gates.CU1Gate`."""
+        from .library.standard_gates.u1 import CU1Gate
+        return self.append(CU1Gate(theta, label=label, ctrl_state=ctrl_state),
+                           [control_qubit, target_qubit], [])
+
+    def mcu1(self, lam, control_qubits, target_qubit):
```

Comment:
i would remove this and add it to generalized gates.

Reply:
yeah all in plan: MCRX, MRCY, MCRZ, MCU1 (should become MCPhase), MCX